components/Chat/ChatInput.vue

```diff
@@ -347,9 +357,41 @@
         }
       },
 
-      // Defer some load events til user interacts with chat
-      async onFocus () {
-        await this.updateEmoteList();
+      updateEmoteList() {
+        this.updateEmoteListInStore();
+
+        if( this.emoteList && this.emoteLinkMap && this.emoteLinkMap.size === this.emoteList.length ) {
+          console.log( 'Skipping emoteList update.' );
+          return;
+        }
+
+        if( !this.emoteLinkMap ) {
+          this.emoteLinkMap = new Map();
+        }
+
+        for( const emote of this.emoteList ) {
+          this.emoteLinkMap.set( emote.image, emote.value );
+        }
+      },
+
+      async onDrop( event ) {
+        if( !this.emoteLinkMap || this.emoteLinkMap.size < 1 ) {
+          await this.updateEmoteList();
+        }
+
+        const droppedText = event.dataTransfer.getData( "text/plain" );
```

Comment:
Will this trigger any errors if non text is pasted into the input field?

Reply:
It's only on drop, and since it's a text-field anyways, I don't see how anything other than text can be dropped (images/videos get linked). Maybe in the future this will hinder formatted pasting.

Reply:
Should be fine.